internal/neofs/neofs.go

```diff
@@ -46,16 +48,29 @@ type NeoFS struct {
 	anonSigner  user.Signer
 	cfg         Config
 	epochGetter EpochGetter
+	buffers     *sync.Pool
 }
 
 // NewNeoFS creates new NeoFS using provided pool.Pool.
 func NewNeoFS(p *pool.Pool, signer user.Signer, anonSigner user.Signer, cfg Config, epochGetter EpochGetter) *NeoFS {
+	buffers := sync.Pool{}
+	buffers.New = func() any {
+		b := make([]byte, cfg.MaxObjectSize)
+		return &b
+	}
+
+	for i := 0; i < cfg.UploadChunkPoolLength; i++ {
+		b := make([]byte, cfg.MaxObjectSize)
+		buffers.Put(&b)
```

Comment:
Not really needed (along with `UploadChunkPoolLength`), it'll be allocated and released dynamically.